tests/foreman/ui/test_environment.py

```diff
@@ -34,17 +36,13 @@ def test_create_env_positive_1(self, name):
 
         """
         with Session(self.browser) as session:
-            make_env(session, name=name)
-            search = self.environment.search(name)
-            self.assertIsNotNone(search)
-
-    @data(
-        gen_string('alpha', 255),
-        gen_string('numeric', 255),
-        gen_string('alphanumeric', 255)
-    )
```

Comment:
* Changing valid_data_list() with more logic to handle removing of specific strings is beyond the scope of this pr.
* Updating valid_env_names() is a good suggestion.  I like it.  But the problem is: if I add it to valid_env_names() then the same value gets passed to the delete test too.  
Unfortunately environment.delete() uses self.delete_entity() from base.py -> which uses self.search_entity() from base.py to search the value.  Searching the long string will fail here because environment.search() has overridden the search feature but delete uses self.search_entity() from base.  This requires search improvement logics which you are working on.  This may be improved when your search improvement goes in but anything I do now will be a temporary patch and will not add any value.